studio/components/interfaces/TableGridEditor/SidePanelEditor/ForeignKeySelector/ForeignKeySelector.utils.tsx

```diff
@@ -0,0 +1,46 @@
+import { FOREIGN_KEY_DELETION_ACTION } from 'data/database/database-query-constants'
+
+export const generateDeletionActionDescription = (deletionAction: string, reference: string) => {
+  switch (deletionAction) {
+    case FOREIGN_KEY_DELETION_ACTION.NO_ACTION:
+      return (
+        <>
+          Deleting a record from <code className="text-xs text-scale-1100">{reference}</code> will{' '}
+          <span className="text-brand-900 opacity-75">raise an error</span> if there exists
+          referencing rows from this table
+        </>
+      )
+    case FOREIGN_KEY_DELETION_ACTION.CASCADE:
+      return (
+        <>
+          Deleting a record from <code className="text-xs text-scale-1100">{reference}</code> will{' '}
+          <span className="text-brand-900 opacity-75">also delete</span> the referencing records
+          from this table
+        </>
+      )
+    case FOREIGN_KEY_DELETION_ACTION.RESTRICT:
+      return (
+        <>
+          Deleting a record from <code className="text-xs text-scale-1100">{reference}</code> will{' '}
+          <span className="text-brand-900 opacity-75">prevent deletion</span> of existing
+          referencing rows from this table
```

Comment:
This is essentially the same as "no action", except it's not deferrable, which is something I don't think most users will care about. What do you think we should be saying here to reflect this?

https://www.postgresql.org/docs/current/sql-createtable.html

Reply:
I think we should put in a little bit of guidance text at the bottom here to help with the confidence factor. What if I don't know which one I want? What if I don't care at this time? What do most people choose? Can I change it later? Can we add a small section to a docs page to cover this in more detail?

![CleanShot 2023-03-07 at 12 39 09@2x](https://user-images.githubusercontent.com/105593/223480194-471d8905-876a-4b5c-8c8a-ccbb7509e25d.png)

Reply:
agreed - i was pretty stuck on the explanation here tbh. cause staging "not deferrable" isn't clear as well for devs new to SQL, lemme think about this a little more

Reply:
great shout on the info box for deciding which action to select, updated as such with an expandable info box:

![image](https://user-images.githubusercontent.com/19742402/223648461-d6244156-f5ac-45cd-adbe-4a02db930f36.png)

![image](https://user-images.githubusercontent.com/19742402/223648412-e99101e2-64b9-45c3-8e30-38520e200423.png)

Reply:
no worries about the docs part - these are postgres specific information, so i'm linking to the docs from postgres here
the info in this infobox is a paraphrasing of what they have on their docs

Reply:
opting to show a tooltip to explain the difference between restrict and no action, trying not to have even more information up front on the UI than we already have - it's getting a bit too wordy and the UX may be degraded

![image](https://user-images.githubusercontent.com/19742402/223650925-29e76a23-3ffa-4e18-a5c4-39803ca6a5a6.png)